### ISSUE TITLE
z/VM 630vm.ins IPL leads to segfault 

### ISSUE THREAD

**Issue by fbi-ranger** [hercules-390/hyperion#136]
IPL 630vm.ins (Installation IPL) leads to segmentation fault in current version of Hyperion. 

HerculesOS: openSUSE 13.2 

This is a regression as the IPL works. 
No additional infromation in hercules.log. 

HHC00414I 0:6201 CKD file /local/sys1/s390/dasd/VM62R1.6201: cyls 10017 heads 15 tracks 150255 trklen 56832                     
HHC00414I 0:6202 CKD file /local/sys1/s390/dasd/VM62C1.6202: cyls 10017 heads 15 tracks 150255 trklen 56832                     
HHC00414I 0:6203 CKD file /local/sys1/s390/dasd/VM62S1.6203: cyls 10017 heads 15 tracks 150255 trklen 56832                     
HHC00414I 0:6204 CKD file /local/sys1/s390/dasd/VM62P1.6204: cyls 10017 heads 15 tracks 150255 trklen 56832                     
HHC00414I 0:6205 CKD file /local/sys1/s390/dasd/VM62T1.6205: cyls 10017 heads 15 tracks 150255 trklen 56832                     
HHC00100I Thread id 00007fd903ae5700, prio  0, name Control panel started                                                       
HHC02915I client 0 COMM: Connection received                                                                                    
HHC02914I 0:0000 COMM: client 0 negotiations complete; ttype = 'IBM-3279-4-E@0000'                                              
HHC01018I 0:0000 COMM: client 127.0.0.1 devtype 3270: connected                                                                 
HHC02915I client 1 COMM: Connection received                                                                                    
HHC02914I 0:001F COMM: client 1 negotiations complete; ttype = 'IBM-3279-4-E@001F'                                              
HHC01018I 0:001F COMM: client 127.0.0.1 devtype 3270: connected                                                                 
HHC01603I ipl /local/sys1/install/CPDVD/630vm.ins                                                                               
HHC00811I Processor CP03: architecture mode ESA/390                                                                             
HHC00811I Processor CP00: architecture mode ESA/390                                                                             
HHC00811I Processor CP01: architecture mode ESA/390                                                                             
HHC00811I Processor CP04: architecture mode ESA/390                                                                             
HHC00811I Processor CP05: architecture mode ESA/390                                                                             
HHC00811I Processor CP02: architecture mode ESA/390                                                                             
HHC00814I Processor CP00: SIGP Set architecture mode            (12) CP00, PARM 00000001: CC 0                                  
HHC00811I Processor CP00: architecture mode z/Arch                                                                              
herc =====> /usr/local/bin/iml-hercules: line 2:  4400 Segmentation fault      /local/sys1/s390/hercules/bin/hercules -f /etc/hercules.cnf -b /local/sys1/s390/etc/herclogo.txt.....Z                                 instcnt 61,295,731; mips 7.236; I/O      0

**Comment by Peter-J-Jansen** [comment]
Florian,

 

Under which Hyperion version did this occur ? Did you use the “-O3” compiler option to generate it ?

 

I never experienced segmentation faults as I always use the default compiler options, both under Windows and Linux, but I’ve read similar error reports and this not-so-trustworthy “-O3” compiler always gets mentioned.

 

I responded to your query because z/VM 6.3 (as you mention 630vm.ins) up until yesterday would not install or run under Hyperion. But yesterday I committed a fix to Hyperion so that z/VM 6.3 now should work just fine, at least it does for me now, after a clean rebuilding with the default compiler settings. I’d be interested in your experiences with that latest Hyperion version.

 

Peter J.
